cdap-cli/src/main/java/co/cask/cdap/cli/CLIConfig.java

```diff
@@ -155,25 +180,30 @@ private AccessToken acquireAccessToken(ClientConfig clientConfig, ConnectionConf
     }
 
     try {
-      AccessToken savedAccessToken = getSavedAccessToken(connectionInfo.getHostname());
-      checkConnection(clientConfig, connectionInfo, savedAccessToken);
-      return savedAccessToken;
+      UserAccessToken savedToken = getSavedAccessToken(connectionInfo.getHostname());
+      connectionInfo = new AuthenticatedConnectionConfig(connectionInfo, savedToken.getUsername());
+      clientConfig.setConnectionConfig(connectionInfo);
+      checkConnection(clientConfig, savedToken.getAccessToken());
+      return savedToken.getAccessToken();
     } catch (UnauthorizedException ignored) {
       // access token invalid - fall through to try acquiring token manually
     }
 
-    return getNewAccessToken(connectionInfo, output, debug);
+    return getNewAccessToken(clientConfig, output, debug);
```

Comment:
I need the connectionInfo to be modified here. Java uses pass by value so any changes I make in the function getNewAccessToken to the connectionConfig won't be saved unless I can set the connectionConfig via clientConfig. (specifically here: connectionInfo = new AuthenticatedConnectionConfig(connectionInfo, username);)